oak_functions_launcher/src/lookup.rs

```diff
@@ -33,17 +34,23 @@ impl UpdateClient<'_> {
     async fn start(&mut self) -> anyhow::Result<()> {
         // Send the current chunk first.
         let update_response = self.send_request(UpdateAction::Start).await?;
-
         if UpdateStatus::Started != update_response.update_status() {
             return Err(anyhow!("Did not receive expected update status: Started"));
         };
+        self.continue_().await
+    }
 
-        // Currently send only data which fits in two chunks.
-        // TODO(#3718): to send more chunks.
-        if self.chunks.len() > 1 {
-            return Err(anyhow!("Cannot send lookup data with more than 2 chunks"));
+    #[async_recursion]
```

Comment:
I am not sure we want to get into recursion territory; both because async recursion is not super stable yet, but also because this recursion is potentially very deep (and may end up using a lot of stack). We can hope that the compiler will tail-optimize these calls, but that's never a guarantee anyways. I suggest an iterative approach instead. It also seems conceptually simpler to me, since semantically we want to iterate over chunks from what is already an iterator, send them one by one, and then finish (with the added slight complication that the last chunk and finish are combined together in the current API design).

Reply:
You are right, done, thank you!